robot-server/robot_server/runs/run_store.py

```
@@ -32,12 +35,64 @@ def __init__(self, run_id: str) -> None:
 class RunStore:
     """Methods for storing and retrieving run resources."""
 
-    def __init__(self) -> None:
+    def __init__(self, sql_engine: sqlalchemy.engine.Engine) -> None:
         """Initialize a RunStore and its in-memory storage."""
-        self._runs_by_id: Dict[str, RunResource] = {}
+        self._sql_engine = sql_engine
 
-    def upsert(self, run: RunResource) -> RunResource:
-        """Insert or update a run resource in the store.
+    def update_actions(self, run_id: str, actions: List[RunAction]) -> None:
+        """Insert or update a run actions resource in the db.
+
+        Arguments:
+            run_id: current run id to get
+            actions: a list of actions to store in the db
+        """
+        try:
+            with self._sql_engine.begin() as transaction:
+                transaction.execute(
+                    sqlalchemy.delete(action_runs_table).where(
+                        action_runs_table.c.run_id == run_id
+                    )
+                )
+        except sqlalchemy.exc.NoResultFound as e:
+            raise e
```

For a few reasons, I don't think this `try ... except` is meaningful.

1. It's re-raising the thing that it caught (`raise e`), which is a no-op.
2. Nothing in the `try` block will raise `NoResultFound`. Note that the `execute()` call will simply affect 0 rows if we're provided a bad `run_id`, not raise `NoResultFound`.
3. If I'm reading this right, it's *expected* for the result set to affect 0 rows sometimes, because a valid run may have 0 actions attached to it.

---

Do we want to raise an error if the affected rows is 0? or return a boolean to mark a successful action?